src/controller.rs

```diff
@@ -253,3 +351,337 @@ where
         _ => Err(Error::JsonPathExactlyOneValue(from_field_path.to_owned())),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::resources::{Mapping, ResourceSyncSpec};
+    use kube::core::{ApiResource, GroupVersionKind};
+
+    #[tokio::test]
+    async fn test_get_ar_from_subtree() {
```

Comment:
hmm i think this is a confusing test, i should just use a configmap. by using a sinker container the reader might expect it to be pulling it out of the inner resource but that's not what it's supposed to do. will fix.